sdks/python/apache_beam/transforms/environments.py

```diff
@@ -109,12 +113,25 @@ class Environment(object):
   _urn_to_env_cls = {}  # type: Dict[str, type]
 
   def __init__(self,
-               capabilities,  # type: Iterable[str]
-               artifacts,  # type: Iterable[beam_runner_api_pb2.ArtifactInformation]
-              ):
+      capabilities,  # type: Iterable[str]
+      artifacts,  # type: Iterable[beam_runner_api_pb2.ArtifactInformation]
+      resource_hints,  # type: Optional[Mapping[str, bytes]]
+               ):
     # type: (...) -> None
     self._capabilities = capabilities
     self._artifacts = artifacts
+    self._resource_hints = dict(resource_hints) if resource_hints else {}
+
+  def __eq__(self, other):
```

Comment:
artifacts? (Capabilities is probably constant...)

Reply:
Yes, I suspected this needs to be added. My change kept preexisting logic. Can we make/expect `artifacts` to be an ordered list or what's the best way to compare them fast? taking a loog through the code to see the actual content & how it is used.

Reply:
Done